core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -346,9 +346,11 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         final RoutingContext routingCtx = req.routingContext();
         final RoutingStatus routingStatus = routingCtx.status();
         if (!routingStatus.routeMustExist()) {
-            final ServiceRequestContext reqCtx = newEarlyRespondingRequestContext(
+            final DefaultServiceRequestContext reqCtx = newEarlyRespondingRequestContext(
                     channel, req, proxiedAddresses, clientAddress, remoteAddress, localAddress, routingCtx);
 
+            reqCtx.hook(config.contextHook());
```

Comment:
We can remove this line because the request is not exposed to users and no context is pushed.